src/Aquifer.Public.API/Endpoints/Bibles/Texts/Get/Validator.cs

```diff
@@ -0,0 +1,53 @@
+﻿using Aquifer.Common.Utilities;
+using Aquifer.Data.Enums;
+using FastEndpoints;
+using FluentValidation;
+
+namespace Aquifer.Public.API.Endpoints.Bibles.Texts.Get;
+
+public class Validator : Validator<Request>
+{
+    public Validator()
+    {
+        RuleFor(x => x.BookCode).NotEmpty();
+        RuleFor(x => x.BookCode)
```

Comment:
I think this can be simplified by putting `.NotEmpty()` on this one and remove the `.When`, because we're already saying it can't be null. The `.When` isn't a valid case.

Reply:
That doesn't work unless I set the default value for `BookCode` on the request to `""` instead of `null!`.  Is that what you're thinking?

Reply:
I'm not sure what you mean. `NotEmpty()` checks against null. It shouldn't go into the `Must` if it's null or empty string. So I think this is all you need:
```csharp
RuleFor(x => x.BookCode).NotEmpty()
            .Must(code => BibleBookCodeUtilities.IdFromCode(code) != BookId.None)
            .WithMessage(x => "Invalid '{PropertyName}': \"{PropertyValue}\".");
```

Reply:
Never mind, I'm wrong about this.

Reply:
I figured out two ways to do the rule chaining.  Given
```
    public string BookCode { get; init; } = null!;
```
both of these work on my local:

1. [Set the cascade mode to stop](https://docs.fluentvalidation.net/en/latest/cascade.html):
```
        RuleFor(x => x.BookCode)
            .Cascade(CascadeMode.Stop)
            .NotEmpty()
            .Must(code => BibleBookCodeUtilities.IdFromCode(code) != BookId.None)
            .WithMessage(x => "Invalid '{PropertyName}': \"{PropertyValue}\".");
```

2. Use [Dependent Rules](https://docs.fluentvalidation.net/en/latest/dependentrules.html) (but the author of the library prefers using the cascade mode):
```
        RuleFor(x => x.BookCode)
            .NotEmpty()
            .DependentRules(() => RuleFor(x => x.BookCode)
                .Must(code => BibleBookCodeUtilities.IdFromCode(code) != BookId.None)
                .WithMessage(x => "Invalid '{PropertyName}': \"{PropertyValue}\"."));
```

Reply:
I went with `CascadeMode.Stop`: [commit](https://github.com/BiblioNexusStudio/aquifer-server/pull/443/commits/915feb0d16be4d3215c483cc56b3e3975e5ae498).